source/js/RecipeExpand.js

```diff
@@ -0,0 +1,496 @@
+        if (data['@graph'][i]['@type'] == 'Recipe') {
+          if (data['@graph'][i]['name']) return data['@graph'][i]['name'];
+        };
+      }
+    }
+    return null;
+  }
+  
+  /**
+   * Extract the yield of the recipe from the given recipe schema JSON obejct
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the recipe yield
+   */
+  function getYield(data) {
+    if (data.recipeYield) return data.recipeYield;
+    if (data['@graph']) {
+      for (let i = 0; i < data['@graph'].length; i++) {
+        if (data['@graph'][i]['@type'] == 'Recipe') {
+          if (data['@graph'][i]['recipeYield']) {
+            if (Array.isArray(data['@graph'][i]['recipeYield'])) {
+              return data['@graph'][i]['recipeYield'][0];
+            } else if (typeof data['@graph'][i]['recipeYield'] == 'string') {
+              return data['@graph'][i]['recipeYield'];
+            }
+          }
+        }
+      }
+    }
+    return null;
+  }
+  
+  /**
+   * Extract the categories of the recipe from the given recipe schema JSON obejct
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the recipe categories as a string
+   */
+  function getCategories(data) {
+    let categories = null;
+    if (data.recipeCategory) {
+      categories = data.recipeCategory
+    } else if (data['@graph']) {
+      for (let i = 0; i < data['@graph'].length; i++) {
+        if (data['@graph'][i]['@type'] == 'Recipe') {
+          if (data['@graph'][i]['recipeCategory']) {
+            categories = data['@graph'][i]['recipeCategory'];
+          }
+        };
+      }
+    }
+    if (Array.isArray(categories)) categories = categories.join(', ');
+    return categories.toLowerCase();
+  }
+  
+  /**
+   * Extract the description of the recipe from the given recipe schema JSON obejct
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the recipe description
+   */
+  function getDescription(data) {
+    if (data.description) return data.description;
+    if (data['@graph']) {
+      for (let i = 0; i < data['@graph'].length; i++) {
+        if (data['@graph'][i]['@type'] == 'Recipe') {
+          return data['@graph'][i]['description'];
+        }
+      }
+    }
+    return null;
+  }
+  
+  /**
+   * Extract a usable image from the given recipe schema JSON object
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the URL of the image as a string, otherwise null
+   */
+  function getImage(data) {
+    if (data.image?.url) return data.image.url;
+    if (data.image?.contentUrl) return data.image.contentUrl;
+    if (data.image?.thumbnail) return data.image.thumbnail;
+    if (data['@graph']) {
+      for (let i = 0; i < data['@graph'].length; i++) {
+        if (data['@graph'][i]['@type'] == 'ImageObject') {
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 6 spaces but found 8.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8447291)
```suggestion
      if (data['@graph'][i]['@type'] == 'ImageObject') {
```